book/tutorials/photon_classifier.py

```diff
@@ -0,0 +1,243 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.16.2
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Machine Learning with ICESat-2 data
+#
+# A machine learning pipeline from point clouds to photon classifications.
+#
+# Reimplementation of
+# https://github.com/YoungHyunKoo/IS2_ML/blob/main/01_find_overlapped_data.ipynb
+
+# %% [markdown]
+# ```{admonition} Learning Objectives
+# By the end of this tutorial, you should be able to:
+# - Convert ICESat-2 point cloud data into an analysis/ML-ready format
+# - Recognize the different levels of complexity of ML approaches and the
+#   benefits/challenges of each
+# - Learn the potential of using ML for ICESat-2 photon classification
+# ```
+
+# %% [markdown]
+# ## Part 0: Setup
+
+# %%
+import earthaccess
+import geopandas as gpd
+import h5py
+import torch
+
+# %% [markdown]
+# ## Part 1: Convert ICESat-2 data into ML-ready format
+#
+# Steps:
+# - Get ATL07 data using [earthaccess](https://earthaccess.readthedocs.io)
+# - Filter to only strong beams
+# - Subset to 6 data variables only
+#
+# TODO: copy Table 1 from Koo et al., 2023 paper
+
+# %%
+# Authenticate using NASA EarthData login
+auth = earthaccess.login()
+s3 = earthaccess.get_s3fs_session(daac="NSIDC")  # Start an AWS S3 session
```

Comment:
Continuing from https://github.com/ICESAT-2HackWeek/website-2024/pull/17#discussion_r1708123531, this is the current error:

```python-traceback
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 3
      1 # Authenticate using NASA EarthData login
      2 auth = earthaccess.login()
----> 3 s3 = earthaccess.get_s3fs_session(daac="NSIDC")  # Start an AWS S3 session

File ~/micromamba/envs/hackweek/lib/python3.11/site-packages/earthaccess/api.py:352, in get_s3fs_session(daac, provider, results)
    350         session = earthaccess.__store__.get_s3fs_session(endpoint=endpoint)
    351         return session
--> 352 session = earthaccess.__store__.get_s3fs_session(daac=daac, provider=provider)
    353 return session

AttributeError: 'NoneType' object has no attribute 'get_s3fs_session'
```

Not user if there's a way to pass auth credentials to GitHub Actions and/or Netlify build so that this line works?

Reply:
Can definitely add secrets for `EARTHDATA_USERNAME` and `EARTHDATA_PASSWORD`, which would enable `earthdata` to login during actions. However GitHub actions run on azure, so s3 access isn't available. I think we'd have to build a self-hosted runner that deploys on AWS.

Reply:
Ideally we could execute these notebooks *on a remotely booted CryoCloud user server*. I think all the pieces exist to do this nowadays? @yuvipanda I saw this relevant issue https://github.com/NASA-IMPACT/veda-jupyterhub/issues/46 as I was exploring your latest awesome 2i2c tech including https://github.com/yuvipanda/jupyter-sshd-proxy. Is there a straightforward way to *start* a user server via CI? Then you'd just have to fire up an ssh connection, execute a notebook, copy and commit the rendered version.

Reply:
@weiji14 For now I recommend executing the notebook on CryoCloud, saving with outputs and adding an entry to not execute it in CI here:

https://github.com/ICESAT-2HackWeek/website-2024/blob/3958f0273e208c502343552ffd23f11b73ac4beb/book/_config.yml#L38-L39